File: linkedin-frontend/src/app/model/pagination/Pagination.ts
import {Inject, Injectable} from "@angular/core";

export enum SortDirection{
  ASC = "ASC",
  DESC = "DESC",
}

@Injectable({
  providedIn: 'root'
})
export class SortDirectionUtil{
  change(sortDirection: SortDirection): SortDirection {
    if(sortDirection === SortDirection.ASC){
      sortDirection = SortDirection.DESC;
    } else {
      sortDirection = SortDirection.ASC;
    }
    return sortDirection;
  }
}

export class Pagination{
  pageSize: number;
  pageNumber: number;
  sortDirection: SortDirection;

  constructor(pageSize: number, pageNumber: number, sortDirection: SortDirection) {
    this.pageSize = pageSize;
    this.pageNumber = pageNumber;
    this.sortDirection = sortDirection;
  }
}

File: modules/postgresdb/src/main/java/com/gbyzzz/linkedinjobsbot/modules/postgresdb/entity/pagination/Pagination.java
package com.gbyzzz.linkedinjobsbot.modules.postgresdb.entity.pagination;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Pagination {
    private Integer pageSize;
    private Integer pageNumber;
    private SortDirection sortDirection;

    public enum SortDirection {
        ASC, DESC
    }
}